TLM/TMPE.CitiesGameBridge/Service/NetService.cs

```diff
@@ -10,83 +10,178 @@ public class NetService : INetService {
 
         private NetService() { }
 
+        // NODE BASICS --------------------------------------------------------------------------------
+
+        /// <summary>
+        /// Check if a node id is valid.
+        /// </summary>
+        /// 
+        /// <param name="nodeId">The id of the node to check.</param>
+        /// 
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
+        public bool IsNodeValid(ushort nodeId) {
+            return CheckNodeFlags(
+                nodeId,
+                NetNode.Flags.Created | NetNode.Flags.Collapsed | NetNode.Flags.Deleted,
+                NetNode.Flags.Created);
+        }
+
+        /// <summary>
+        /// Check node flags contain at least one of the flags in <paramref name="flagMask"/>.
+        /// </summary>
+        /// 
+        /// <param name="nodeId">The id of the node to inspect.</param>
+        /// <param name="flagMask">The flags to test.</param>
+        /// <param name="expectedResult">If specified, ensure only the expected flags are found.</param>
+        /// 
+        /// <returns>Returns <c>true</c> if the test passes, otherwise <c>false</c>.</returns>
+        public bool CheckNodeFlags(ushort nodeId,
```

Comment:
I just kept original naming for now.

In future I plan to completely replace the whole `GameBridge` stuff and get some new properly named and documented 'connector' classes added.

With that proposed new approach, the connectors will become members of any class that needs them, so checking node flags would become something like:

```cs
Node.HasFlag(nodeId, flag)

Node.HasAnyFlag(nodeId, flag | flag ...)
```

So for now I think leave old naming, as it will be getting completely replaced when the `GameBridge` is replaced.

Reply:
I recall you were complaining `CheckFlags` with flag mask and expected result is stupid and you could easily do `flags & (mask) == expected`

Reply:
Yup, eventually it will all go. For now my focus was just evicting the LogicUtil class.